File: Omega/src/com/saggitt/omega/search/webproviders/BaiduWebSearchProvider.java
/*
 *  Copyright (c) 2020 Omega Launcher
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 */

package com.saggitt.omega.search.webproviders;

import android.content.Context;
import android.graphics.drawable.Drawable;

import com.android.launcher3.R;

import org.jetbrains.annotations.NotNull;

public class BaiduWebSearchProvider extends WebSearchProvider {
    public BaiduWebSearchProvider(@NotNull Context context) {
        super(context);
        searchUrl = "https://m.baidu.com/s?word=%s";
        suggestionsUrl = "https://m.baidu.com/su?action=opensearch&ie=UTF-8&wd=%s";
        name = mContext.getResources().getString(R.string.web_search_baidu);
    }

    @NotNull
    @Override
    public Drawable getIcon() {
        return mContext.getResources().getDrawable(R.drawable.ic_baidu);
    }
}

File: Omega/src/com/saggitt/omega/search/webproviders/BaiduWebSearchProvider.kt
/*
 * This file is part of Neo Launcher
 * Copyright (c) 2023   Neo Launcher Team
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package com.saggitt.omega.search.webproviders

import android.content.Context
import android.graphics.drawable.Drawable
import androidx.core.content.res.ResourcesCompat
import com.android.launcher3.R
import com.saggitt.omega.search.WebSearchProvider

class BaiduWebSearchProvider(context: Context) : WebSearchProvider(context) {
    override val iconRes: Int
        get() = R.drawable.ic_baidu
    override val icon: Drawable
        get() = ResourcesCompat.getDrawable(context.resources, iconRes, null)!!

    override val packageName: String
        get() = "https://www.baidu.com/s?wd=%s"

    override val suggestionsUrl: String
        get() = "https://m.baidu.com/su?action=opensearch&ie=UTF-8&wd=%s"

    override val displayName: String
        get() = context.resources.getString(R.string.web_search_baidu)
}